packages/amplify-category-api/src/provider-utils/awscloudformation/utils/manage-api-key.ts

```diff
@@ -0,0 +1,71 @@
+import { $TSContext, stateManager } from 'amplify-cli-core';
+import { AppSyncClient, ListApiKeysCommand, UpdateApiKeyCommand } from '@aws-sdk/client-appsync';
+import { loadConfigurationForEnv } from 'amplify-provider-awscloudformation';
+import { contextUtil } from '../../../category-utils/context-util';
+
+const SECONDS_PER_DAY = 86400;
+
+export interface ApiKeyStatus {
+  expiration?: number;
+  key?: string;
+  apiId?: string;
+  description?: string;
+  exists: boolean;
+}
+
+const getAppSyncClient = async (context: $TSContext): Promise<AppSyncClient> => {
```

Comment:
Is it possible to get the appsync client from `amplify-provider-awscloudformation` package as I see most of the client calls are defined there?

Reply:
It doesn't currently expose anything for this, and we don't want to add too many exposed APIs in that package, this is just me de-duping code in the file